compiler-cli/src/dependencies.rs

```diff
@@ -691,18 +708,55 @@ fn resolve_versions<Telem: Telemetry>(
     }
 
     // Convert provided packages into hex packages for pub-grub resolve
-    let provided_hex_packages = provided_packages
+    let provided_hex_packages: HashMap<EcoString, hexpm::Package> = provided_packages
         .iter()
         .map(|(name, package)| (name.clone(), package.to_hex_package(name)))
         .collect();
 
-    let resolved = dependency::resolve_versions(
+    let root_requirements_clone = root_requirements.clone();
```

Comment:
Looking for suggestions of better designs for this section. Cloning for use in calling the function recursively on the `should_try_unlock` page feels a bit gross